### PR TITLE
Migrate to eslint for linting.

### PR DESCRIPTION
**Pull request by Davidhanson90**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    overrides: [
+        {
+            files: ['*.js'],
+            rules: {
+                '@typescript-eslint/no-var-requires': 'off',
+            },
+        },
+        {
+            files: ['*.ts'],
+            rules: {
```
Comment (Roaders) [inline .eslintrc.js:33]: I think these should just go in rules, not in overrides

File: prettier.config.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
-    trailingComma: "all",
+    trailingComma: 'all',
```
Comment (Roaders) [inline prettier.config.js:2]: if you want to keep this file you will have to require it at the top of eslintrc.js and then use the config in place of the prettier config in there already. Alternatively you can just copy this config into eslintrc and delte this

**Review by Roaders — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (Davidhanson90) [reply .eslintrc.js]: Ok moved.

File: prettier.config.js (hunk 2 above)
Comment (Davidhanson90) [reply prettier.config.js:2]: Not sure I understand here, my file is being respected with out any import.  If I remove the file it forces a load of format changes I don't want.

Comment (Davidhanson90) [reply prettier.config.js:2]: Aligned prettier to the root config.

**Review by Roaders — APPROVED** [review]

**Pull request merged**